scripts/test-many-configs.sh

```diff
@@ -0,0 +1,419 @@
+#!/bin/bash
```

Comment:
Should we put a note that this script should only ever be run on the GitHub Actions CI, so one should look at the corresponding yml files to see how to use this script?